### PR TITLE
 Use three-valued logic with missing values in ==(::Tuple, ::Tuple) 

### PR DESCRIPTION
**Pull request by nalimilan**
For consistency with `AbstractArray` (missed by https://github.com/JuliaLang/julia/pull/24653).

Also add a test for the special case when the result of equality test is not `missing` even in the presence of missing values.

### CODE REVIEW

**Comment by nalimilan** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by KristofferC** [comment]
Why not use another operator to opt in to 3VL? If this is derailing the discussion, I can ask wherever it is more suitable.

File: test/missing.jl [hunk 1]
```diff
@@ -217,11 +218,26 @@ end
...
     @test [1] != Union{Int, Missing}[2]
     @test Union{Int, Missing}[1] != Union{Int, Missing}[2]
 end
 
+@testset "== and != on tuples" begin
+    @test ismissing((1, missing) == (1, missing))
+    @test ismissing(("a", missing) == ("a", missing))
+    @test ismissing((missing,) == (missing,))
+    @test ismissing((missing, 2) == (1, missing))
+    @test !((missing, 1) == (missing, 2))
```
Comment (ararslan) [inline test/missing.jl]: I could be missing something super obvious, but how is this different than the test a few lines down that does `!=`?

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/24963#issuecomment-349983001) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/c7d3938249459ee7f85533b7d35ab4da20a3c34e/a180fc7_vs_0b34071/report.md). cc @ararslan

File: test/missing.jl (hunk 1 above)
Comment (nalimilan) [reply test/missing.jl]: Yeah, it's the same since the fallback is defined using `==`, but that's the assumption we're checking. :-)

**Comment by nalimilan** [comment]
OK, the "simd/local_arrays" regression cannot possibly be related. The "tuple/reduction" benchmark doesn't use `==` either. The "dates" regressions are also noisy benchmarks -- and so are the improvements. So overall I think we can conclude that nothing changed. Anyway if there was a type unstability, it would have a really dramatic impact.

**Comment by nalimilan** [comment]
I'll merge tomorrow barring objections.

**Comment by StefanKarpinski** [comment]
LGTM: the scheme is that `isequal` always returns a boolean and `==` uses 3VL, right?

**Comment by nalimilan** [comment]
Yes. BTW, I've noticed `<` and `>` need a similar change (with `isless` always returning `Bool` like `isequal`).

**Comment by StefanKarpinski** [comment]
Circle CI failures look legit and related, btw.

**Comment by nalimilan** [comment]
> Circle CI failures look legit and related, btw.

No, AFAICT that's https://github.com/JuliaLang/julia/pull/24984. Let's see what a new run says.

**Comment by nalimilan** [comment]
CircleCI still fails, but with `ProcessExitedException`. Merging.

**Pull request merged**